api/renter.go

```diff
@@ -275,17 +331,57 @@ func (api *API) renterUploadHandler(w http.ResponseWriter, req *http.Request, ps
 		return
 	}
 
+	// Check whether the erasure coding parameters have been supplied.
+	var ec modules.ErasureCoder
+	if req.FormValue("datapieces") != "" || req.FormValue("paritypieces") != "" {
+		// Check that both values have been supplied.
+		if req.FormValue("datapieces") == "" || req.FormValue("paritypieces") == "" {
+			WriteError(w, Error{"Must provide both the datapieces paramaeter and the paritypieces parameter if specifying erasure coding parameters"}, http.StatusBadRequest)
+			return
+		}
+
+		// Parse the erasure coding parameters.
+		var dataPieces, parityPieces int
+		_, err := fmt.Sscan(req.FormValue("datapieces"), &dataPieces)
+		if err != nil {
+			WriteError(w, Error{"Unable to read parameter 'datapieces': " + err.Error()}, http.StatusBadRequest)
+			return
+		}
+		_, err = fmt.Sscan(req.FormValue("paritypieces"), &parityPieces)
+		if err != nil {
+			WriteError(w, Error{"Unable to read parameter 'paritypieces': " + err.Error()}, http.StatusBadRequest)
+			return
+		}
+
+		// Verify that sane values for parityPieces and redundancy are being
+		// supplied.
+		if parityPieces < requiredParityPieces {
+			WriteError(w, Error{fmt.Sprintf("A minimum of %v parity pieces is required, but %v parity pieces requested.", parityPieces, requiredParityPieces)}, http.StatusBadRequest)
```

Comment:
I don't think we add context to our api errors. But it's something I could see us doing in the future.

Open to explore this more, but during this PR I was assuming that the api error would be directly fed to the user.